internal/v1/handler_oscap.go

```diff
@@ -205,126 +207,548 @@ func (h *Handlers) lintOpenscap(ctx echo.Context, cust *Customizations, fixup bo
 	if err != nil {
 		return nil, err
 	}
-	bp, err := h.server.complianceClient.PolicyCustomizations(ctx.Request().Context(), major, minor, policy)
+
+	// Get current policy customizations
+	currentPolicyBp, err := h.server.complianceClient.PolicyCustomizations(ctx.Request().Context(), major, minor, policy)
 	if err == compliance.ErrorTailoringNotFound {
 		return nil, err
 	} else if err != nil {
 		return nil, echo.NewHTTPError(http.StatusInternalServerError, err)
 	}
 
-	// make sure all packages are present, all partitions, all enabled/disabled services, all kernel args
-	for _, pkg := range bp.GetPackagesEx(false) {
-		if cust.Packages == nil || !slices.Contains(*cust.Packages, pkg) {
-			lintErrors = append(lintErrors, BlueprintLintItem{
-				Name:        "Compliance",
-				Description: fmt.Sprintf("package %s required by policy is not present", pkg),
-			})
-			if fixup {
-				cust.Packages = common.ToPtr(append(common.FromPtr(cust.Packages), pkg))
+	// Get saved policy blueprint from service snapshots
+	savedPolicyBp, err := h.getSavedPolicyBlueprint(blueprintEntry, policy)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check for policy additions (items added to current policy vs saved policy)
+	lintErrors = append(lintErrors, h.checkPolicyAdditions(currentPolicyBp, savedPolicyBp, cust, fixup)...)
```

Comment:
It is more straitforward to simply pass a pointer to slice so the method can add it.